src/components/Modal/index.tsx

```diff
@@ -0,0 +1,38 @@
+import { ReactElement, FC } from 'react';
+import styles from './index.module.scss';
+
+interface ModalProps {
+  children: ReactElement;
+  onClose: () => null;
+}
+
+const Modal: FC<ModalProps> = ({ children, onClose }) => {
+  return (
+    <>
+      <Backdrop />
+      <Overlay onClose={onClose}>{children}</Overlay>
```

Comment:
Instead of creating these separate components, add the whole JSX of both of them here only